optuna/study.py

```diff
@@ -457,6 +457,8 @@ def ask(
         trial_id = self._pop_waiting_trial_id()
         if trial_id is None:
             trial_id = self._storage.create_new_trial(self._study_id)
+        else:
+            self._storage.set_trial_state(trial_id, TrialState.RUNNING)
```

Comment:
I think this state update is already performed in `_pop_waiting_trial_id` and isn't needed therefore.

Reply:
You're right. Sorry forgot that